### PR TITLE
Fix TS and ESLint issues in abtest.ts

### PR DESCRIPTION
**Pull request by GHaberis**
## What are you doing in this PR?

Fix TS and ESLint issues in abtest.ts.

### CODE REVIEW

File: support-frontend/assets/helpers/abTests/abtest.ts [hunk 1]
```diff
@@ -200,7 +220,8 @@ function userInTest(
 }
 
 function randomNumber(mvtId: number, seed: number): number {
-	const rng = seedrandom(mvtId + seed);
+	const numberToSeed = mvtId + seed;
+	const rng = seedrandom(numberToSeed.toString());
```
Comment (GHaberis) [inline support-frontend/assets/helpers/abTests/abtest.ts:224]: I've added `@types/seedrandom` and this `seedrandom` function takes a `string`.

File: support-frontend/assets/helpers/abTests/abtest.ts [hunk 2]
```diff
@@ -327,10 +348,6 @@ function getAmountsTestParticipations(
 	countryGroupId: CountryGroupId,
 	settings: Settings,
 ): Participations | null | undefined {
-	if (!settings.amounts) {
```
Comment (GHaberis) [inline support-frontend/assets/helpers/abTests/abtest.ts:330]: This conditional was flagged as needless as the definition for `Settings` (`window.guardian.settings`) always has `window.guardian.settings.amounts` is not optional.

File: support-frontend/assets/helpers/abTests/abtest.ts (hunk 1 above)
Comment (marialani) [reply support-frontend/assets/helpers/abTests/abtest.ts:224]: There's a PR in flight [here](https://github.com/guardian/support-frontend/pull/3324#discussion_r748232270) which modifies this file that also caused the build to fail

Comment (marialani) [reply support-frontend/assets/helpers/abTests/abtest.ts:224]: Specific comment [here](https://github.com/guardian/support-frontend/pull/3324#discussion_r748232270)

Comment (GHaberis) [reply support-frontend/assets/helpers/abTests/abtest.ts:224]: @marialani thanks for drawing my attention to this, I've left a comment

**Pull request closed without merging**